src/pandapipes/diagnostic.py

```diff
@@ -0,0 +1,145 @@
+        else:
+            logger.warning("The initial, unmodified pipeflow does NOT converge.")
+    except Exception as e:
+        logger.info(f"The initial, unmodified pipeflow does NOT converge.\n"
+                    f"\t\tThis exception is raised:\n\t\t{e}")
+
+    # check ext_grid
+    if net.fluid.is_gas & (not hasattr(net, "ext_grid") | net.ext_grid.empty):
+        logger.warning("The net does not have an external grid! "
+                       "An external grid is required for gas networks.")
+
+    # check zero / low length
+    zl = net.pipe.loc[net.pipe.length_km == 0]
+    ll = net.pipe.loc[net.pipe.length_km <= low_length_limit_km]
+    if not zl.empty:
+        logger.warning(f"{len(zl.index)} pipes have a length of 0.0 km. (IDs: {zl.index})")
+    if not ll.empty:
+        logger.warning(f"{len(ll.index)} pipes have a length below"
+                       f" {low_length_limit_km} km. "
+                       f"This could lead to convergence issues. The lowest length in the net is "
+                       f"{ll.length_km.min()} km.\n"
+                       f"(IDs of pipelines with low length: {ll.index})")
+
+    net2 = net.deepcopy()
+    net2.pipe.loc[net2.pipe.length_km < low_length_limit_km].length_km = low_length_limit_km
+    try:
+        pp.pipeflow(net2)
+        if net2.converged:
+            logger.info(f"If all short pipelines (< {low_length_limit_km} km) were set to "
+                        f"{low_length_limit_km} km, the pipeflow would converge.")
+        else:
+            logger.warning(f"If all short pipelines (< {low_length_limit_km} km) were set to "
+                        f"{low_length_limit_km} km, the pipeflow would still NOT converge.")
+    except Exception as e:
+        logger.info(f"Pipeflow does not converge, even if all short pipelines (< 10 m) were set "
+                    f"to {low_length_limit_km} km. \n"
+                    f"\t\tThe error message is: {e}")
+
+    # check iterations
+    iterations = 200
+    try:
+        pp.pipeflow(net, iter=iterations)
+        logger.info(f"The pipeflow converges after {net._internal_results['iterations']:d} "
+                    f"iterations.")
+    except PipeflowNotConverged:
+        logger.info(f"After {iterations:d} iterations the pipeflow did NOT converge.")
+
+    # check with little sink and source scaling
+    logger.info("Testing with scaled-down sinks and sources.")
+    net3 = net.deepcopy()
+    if hasattr(net, "sink"):
+        net3.sink.scaling *= check_scaling_factor
+    if hasattr(net, "source"):
+        net3.source.scaling *= check_scaling_factor
+    try:
+        pp.pipeflow(net3)
+        if net3.converged:
+            logger.info(f"If sinks and sources were scaled with a factor of to "
+                        f"{check_scaling_factor}, the pipeflow would converge.")
+        else:
+            logger.warning(f"If sinks and sources were scaled with a factor of to "
```

Comment:
See comment above.

Reply:
You mean iterations as a function argument?

Reply:
I mean the pipeflow exception that must be caught. The else-part will never be reached.